lib/ansible/parsing/mod_args.py

```diff
@@ -330,3 +333,22 @@ def parse(self, skip_action_validation=False):
                                          obj=self._task_ds)
 
         return (action, args, delegate_to)
+
+
+def get_action_groups_config(action, collection_list):
+    name, action_path = module_loader.find_plugin_with_name(action, collection_list=collection_list, check_aliases=True)
```

Comment:
The second one feels kind of odd. I should expect that collections can use each other's action_groups? That seems like it would be a surprise to me as a user and would rather have things safely namespaced. Why not just inspect the collection of the task action and if it's not there, there is no group? The 1st's syntax to me seems pretty unfortunate but perhaps better than what I've done practically.

I did specifically avoided touching the action/module stuff since this PR was just to fix groupings, but I have looked into it and played with some possible solutions. FQCN works currently, except for ansible.builtin and ansible.legacy, which are just broken.

I asked bcoca about caching the results to improve performance and decided to hold off until I had something working decently. And yes, let's talk about the playbook/DS - the placement of this stuff feels a little awkward.